Week 4/Contest (Apr 3)/Shopping List (Easy)/dominic.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <vector>
+#include <set>
+using namespace std;
+
+int main() {
+  int n, m;
+  cin >> n >> m;
+
+  string s;
+  set<string> res;
+  for (int j = 0; j < m; j++) {
+    cin >> s;
+    res.insert(s);
+  }
+  for (int i = 1; i < n; i++) {
+    set<string> curr;
+    for (int j = 0; j < m; j++) {
+      cin >> s;
+      curr.insert(s);
+    }
+    vector<string> rem;
```

Comment:
you can use std::set_intersection: https://en.cppreference.com/w/cpp/algorithm/set_intersection

if you really want to implement yourself, I'd recommend checking curr and res first before joining them:
```
if (curr.size() < res.size()) swap(curr, res);
```
so we only loop through the small set during the intersection